opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -390,6 +390,26 @@ def values(self) -> typing.ValuesView[str]:
 
 DEFAULT_TRACE_STATE = TraceState.get_default()
 
+_TRACE_ID_BYTES = 16
+_TRACE_ID_HEX_LENGTH = 2 * _TRACE_ID_BYTES
+
+
+def _validate_trace_id(trace_id: int) -> bool:
+    """Validates a trace_id.
+
+    Args:
+        trace_id: An int that is expected to corresponds to 16-bytes hexdecimal value
+
+    Returns:
+        True if trace_id is valid, False otherwise.
+    """
+    if not trace_id:
+        return False
+    if len(format(trace_id, "032x")) != _TRACE_ID_HEX_LENGTH:
+        return False
+
+    return True
```

Comment:
Building off of what @aabmass said. I'm okay with constant/no constant since it's only used once but having a `_MAX_TRACE_ID_LENGTH` is probably easier to read!

```suggestion
    return trace_id and trace_id < (1 << 128) - 1 and trace_id != INVALID_TRACE_ID
```